# HG changeset patch
# User Rik <rik@octave.org>
# Date 1351100766 25200
#      Wed Oct 24 10:46:06 2012 -0700
# Node ID 8e6a72cac999ed3346d3b37d70f7c5ea82301c0f
# Parent  13206177060be3db1b6f55219a7c36bb013c327b
unimplemented.m: Add docstring for internal function.

* unimplemented.m: Add docstring for internal function.

diff --git a/scripts/help/unimplemented.m b/scripts/help/unimplemented.m
--- a/scripts/help/unimplemented.m
+++ b/scripts/help/unimplemented.m
@@ -13,22 +13,34 @@
 ## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 ## General Public License for more details.
 ##
 ## You should have received a copy of the GNU General Public License
 ## along with Octave; see the file COPYING.  If not, see
 ## <http://www.gnu.org/licenses/>.
 
 ## -*- texinfo -*-
-## @deftypefn {Function File} {} unimplemented ()
-## Undocumented internal function.
+## @deftypefn {Function File} {@var{txt} =} unimplemented (@var{fcn})
+## Return specific help text for the unimplemented function @var{fcn}.
+## This is usually a suggestion for an existing compatible function to use in
+## place of @var{fcn}.
+##
+## This function is not called by users, but by the Octave interpreter when
+## it fails to recognize an input string as a valid function name.  See
+## @code{missing_function_hook} for using a different handler for this event.
+## @seealso{missing_function_hook}
 ## @end deftypefn
 
+
 function txt = unimplemented (fcn)
 
+  if (nargin != 1)
+    print_usage ();
+  endif
+
   is_matlab_function = true;
 
   ## Some smarter cases, add more as needed.
   switch (fcn)
 
   case "quad2d"
     txt = ["quad2d is not implemented.  Consider using dblquad."];
 
